frontend/public/components/dashboards-page/overview-dashboard/details-card.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { connect } from 'react-redux';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../dashboard/dashboard-card';
+import { DetailsBody, DetailItem } from '../../dashboard/details-card';
+import { withDashboardResources, DashboardItemProps } from '../with-dashboard-resources';
+import { InfrastructureModel, ClusterVersionModel } from '../../../models';
+import { referenceForModel } from '../../../module/k8s';
+import { FLAGS } from '../../../const';
+import { featureReducerName, flagPending } from '../../../reducers/features';
+import { RootState } from '../../../redux';
+import { getK8sGitVersion, getOpenShiftVersion } from '../../about-modal';
+import { FirehoseResource } from '../../utils';
+
+const getClusterName = (): string => {
+  const apiServerURL = window.SERVER_FLAGS.kubeAPIServerURL;
+  if (!apiServerURL) {
+    return null;
+  }
+  const clusterName = apiServerURL.replace('https://api.', '');
+  const portIndex = clusterName.indexOf(':');
+  if (portIndex !== -1) {
+    return clusterName.slice(0, portIndex);
+  }
+  return clusterName;
+};
+
+const getInfrastructurePlatform = (infrastructure: object): string => _.get(infrastructure, 'status.platform');
+
+const clusterVersionResource: FirehoseResource = {
+  kind: referenceForModel(ClusterVersionModel),
+  namespaced: false,
+  name: 'version',
+  isList: false,
+  prop: 'cv',
+};
+
+const infrastructureResource: FirehoseResource = {
+  kind: referenceForModel(InfrastructureModel),
+  namespaced: false,
+  name: 'cluster',
+  isList: false,
+  prop: 'infrastructure',
+};
+
+const mapStateToProps = (state: RootState) => ({
+  openShiftFlag: state[featureReducerName].get(FLAGS.OPENSHIFT),
+});
+
+export const _DetailsCard: React.FC<DetailsCardProps> = ({
+  watchURL,
+  stopWatchURL,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+  urlResults,
+  openShiftFlag,
+}) => {
+  React.useEffect(() => {
+    if (!flagPending(openShiftFlag)) {
+      if (openShiftFlag) {
+        watchK8sResource(clusterVersionResource);
+        watchK8sResource(infrastructureResource);
+        return () => {
+          stopWatchK8sResource(clusterVersionResource);
+          stopWatchK8sResource(infrastructureResource);
+        };
+      }
+      watchURL('version');
+      return () => {
+        stopWatchURL('version');
+      };
+    }
+  }, [openShiftFlag, watchK8sResource, stopWatchK8sResource, watchURL, stopWatchURL]);
+  const clusterVersionLoaded = _.get(resources, 'cv.loaded', false);
+  const openshiftVersion = getOpenShiftVersion(_.get(resources, 'cv'));
+
+  const infrastructureLoaded = _.get(resources, 'infrastructure.loaded', false);
+  const infrastructureData = _.get(resources, 'infrastructure.data', {});
```

Comment:
```suggestion
  const infrastructureData: K8sResourceKind = _.get(resources, 'infrastructure.data', {});
```

although TypeScript will yell since `{}` isn't a `K8sResourceKind`. We should fix that though so we have the type checking.

Reply:
type added